src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -1771,7 +1783,35 @@ constexpr bool check_mutate_apply_argument_tags(
  * DataBox, `box`
  */
 template <typename MutateTags, typename ArgumentTags, typename F,
-          typename BoxTags, typename... Args>
+          typename BoxTags, typename... Args,
+          Requires<not DataBox_detail::has_return_tags_and_argument_tags_v<F>> =
+              nullptr>
+inline constexpr auto mutate_apply(
+    F&& f, const gsl::not_null<DataBox<BoxTags>*> box,
+    Args&&... args) noexcept(DataBox_detail::
+                                 check_mutate_apply_mutate_tags(
+                                     BoxTags{}, MutateTags{}) and
+                             DataBox_detail::check_mutate_apply_argument_tags(
+                                 BoxTags{}, ArgumentTags{}) and
+                             noexcept(DataBox_detail::mutate_apply(
+                                 f, box, MutateTags{}, ArgumentTags{},
+                                 std::forward<Args>(args)...))) {
+  // These checks are duplicated in the noexcept specification above
+  // because the noexcept(DataBox_detail::mutate_apply(...)) can cause
+  // a compilation error before the checks in the function body are
+  // performed.
+  DataBox_detail::check_mutate_apply_mutate_tags(BoxTags{}, MutateTags{});
+  DataBox_detail::check_mutate_apply_argument_tags(BoxTags{}, ArgumentTags{});
+  return DataBox_detail::mutate_apply(std::forward<F>(f), box, MutateTags{},
+                                      ArgumentTags{},
+                                      std::forward<Args>(args)...);
+}
+
+template <
+    typename F, typename BoxTags, typename... Args,
+    Requires<DataBox_detail::has_return_tags_and_argument_tags_v<F>> = nullptr,
+    typename MutateTags = typename F::return_tags,
+    typename ArgumentTags = typename F::argument_tags>
```

Comment:
Are these last two template parameters needed?

Reply:
@nilsdeppe can you answer?

Reply:
@nilsdeppe says they are used as type aliases that are also visible in the `noexcept` so that we don't have to type them out. no strong opinion on marking it unconditionally `noexcept` or keeping it the way it is.

Reply:
OK.